### PR TITLE
Optimize code of azure-spring-boot-starter-keyvault-certificates

### PR DESCRIPTION
**Pull request by chenrujun**
- Add public static final variable to make code easier to read.
 - Delete duplicated codes.

### CODE REVIEW

**Comment by chenrujun** [comment]
Hi, @saragluna . Please help to review the PR when you have time.

File: sdk/spring/azure-spring-boot-starter-keyvault-certificates/src/main/java/com/azure/spring/security/keyvault/certificates/starter/KeyVaultCertificatesEnvironmentPostProcessor.java [hunk 1]
```diff
@@ -3,112 +3,89 @@
...
-                systemProperties.put("azure.keyvault.client-secret", clientSecret);
-            }
+        if (environment.getProperty("azure.keyvault.uri") == null) {
+            return;
+        }
 
-            String managedIdentity = environment.getProperty("azure.keyvault.managed-identity");
-            if (managedIdentity != null) {
-                systemProperties.put("azure.keyvault.managed-identity", managedIdentity);
+        putEnvironmentPropertyToSystemProperty(environment, "azure.keyvault.aad-authentication-url");
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-starter-keyvault-certificates/src/main/java/com/azure/spring/security/keyvault/certificates/starter/KeyVaultCertificatesEnvironmentPostProcessor.java:34]: better add a comment here to explain why copy the env.prop to system properties.

File: sdk/spring/azure-spring-boot-starter-keyvault-certificates/src/main/java/com/azure/spring/security/keyvault/certificates/starter/KeyVaultCertificatesEnvironmentPostProcessor.java [hunk 2]
```diff
@@ -3,112 +3,89 @@
...
-                    properties.put("server.ssl.key-store-type", "DKS");
-                } catch (ClassNotFoundException ex) {
-                }
-
-                PropertiesPropertySource propertySource
-                        = new PropertiesPropertySource("KeyStorePropertySource", properties);
-                sources.addFirst(propertySource);
+            propertySources.addFirst(new PropertiesPropertySource("KeyStorePropertySource", properties));
+        }
+        if (KeyVaultKeyStore.KEY_STORE_TYPE.equals(environment.getProperty("server.ssl.trust-store-type"))) {
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-starter-keyvault-certificates/src/main/java/com/azure/spring/security/keyvault/certificates/starter/KeyVaultCertificatesEnvironmentPostProcessor.java:50]: need another variable like TRUST_STORE_TYPE?

**Review by saragluna — CHANGES_REQUESTED** [review]

File: sdk/spring/azure-spring-boot-starter-keyvault-certificates/src/main/java/com/azure/spring/security/keyvault/certificates/starter/KeyVaultCertificatesEnvironmentPostProcessor.java (hunk 1 above)
Comment (chenrujun) [reply sdk/spring/azure-spring-boot-starter-keyvault-certificates/src/main/java/com/azure/spring/security/keyvault/certificates/starter/KeyVaultCertificatesEnvironmentPostProcessor.java:34]: Added in the method.

File: sdk/spring/azure-spring-boot-starter-keyvault-certificates/src/main/java/com/azure/spring/security/keyvault/certificates/starter/KeyVaultCertificatesEnvironmentPostProcessor.java (hunk 2 above)
Comment (chenrujun) [reply sdk/spring/azure-spring-boot-starter-keyvault-certificates/src/main/java/com/azure/spring/security/keyvault/certificates/starter/KeyVaultCertificatesEnvironmentPostProcessor.java:50]: No. It's just a key store name. No mater trust or not trust, we can use one store name.

**Comment by chenrujun** [comment]
Hi, @saragluna , @yiliuTo .
Please review this PR.

**Review by yiliuTo — APPROVED** [review]
LGTM

**Comment by chenrujun** [comment]
Closing this PR. The commits in this PR already been cherry-picked in #19947

**Pull request closed without merging**